enterprise/edx_api.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
```

Comment:
So I'm slightly confused by this new module -- I do like the general idea, but I'm trying to understand if this module manages access to clients across *all* Open edX services, or if it's just for the LMS.  Currently it seems we are only concerned with LMS services, in which case we might want to rename the module to reflect the domain.

However, what might be better here is to introduce a `services` package/module which could contain all of the various adapters to communicate with remote services.

@zubair-arbi @saleem-latif @asadiqbal08 FYI